### PR TITLE
MBS-11472 tests for instrumentation local args

### PR DESCRIPTION
**Pull request by dsvoronin**
fix verboseMode being not serializable (failed on worker execution)

dump local args and runner args in build folder for any run

### CODE REVIEW

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/AndroidInstrumentationArgsTester.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.avito.instrumentation.internal
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import java.io.File
+
+internal class AndroidInstrumentationArgsTester(private val dumpDir: File) {
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/AndroidInstrumentationArgsTester.kt]: I didn't get why "testet". Does it test anything?
Maybe "dumper"?

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt [hunk 2]
```diff
@@ -71,12 +72,19 @@ public class InstrumentationTestsPlugin : Plugin<Project> {
 
         project.withAndroidModule { testedExtension ->
 
-            var pluginLevelInstrumentationArgs: Map<String, String> = emptyMap()
+            val dumpDir = File(extension.output, dumpDirName).apply {
+                mkdirs()
+            }
+
+            project.afterEvaluate {
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt]: Do we really have to write files at any project configuration? Why not in the execution phase?

Comment (dsvoronin) [reply subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt]: there is no task to execute for this local run scenario
Android Studio test run doesn't actually run connectedCheck and the last known task is test and main apk's `install`

I don't like it either and it's not configuration cache friendly, but the only way i see it is to hide behind "test" flag as i did before

Comment (eugene-krivobokov) [reply subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt]: Will you do it? Will you make it optional and explicit?

Comment (dsvoronin) [reply subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt]: yes

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/AndroidInstrumentationArgsDumper.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    fun readDump(): Map<String, String> {
+        return gson.fromJson(getDumpFile().reader(), Map::class.java) as Map<String, String>
+    }
+
+    private fun getDumpFile(): File {
+        return File(dumpDir, "local-instrumentation-args-dump.json")
+    }
+}
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/AndroidInstrumentationArgsDumper.kt:22]: This is a new helper for debugging.

Let's start making a troubleshooting guide for the test runner. The simplest guide will be better than ideal in backlog.

I see it's structure as a brief architecture overview (main components) and explaining intermediate artifacts for debugging.

**Comment by sergioboi** [comment]
Won't review there are no remote branch

**Pull request merged**